tasks/main.yml

```diff
@@ -24,9 +24,32 @@
     ( ansible_distribution_major_version > "7" and
       ( ansible_distribution == "CentOS" or ansible_distribution == "RedHat" ))
 
-- name: Set permanent SELinux state
-  selinux: policy={{ selinux_policy }} state={{ selinux_state }}
-  when: selinux_state is defined
+- name: Set permanent SELinux state if enabled
+  selinux:
+    state: "{{ selinux_state | default(ansible_selinux.config_mode) }}"
+    policy: "{{ selinux_policy | default(ansible_selinux.type) }}"
+  register: selinux_mod_output
+  when: ansible_selinux.status == "enabled"
+
+- name: Set permanent SELinux state is disabled
+  selinux:
+    state: "{{ selinux_state }}"
+    policy: "{{ selinux_policy | default('targeted') }}"
+  register: selinux_mod_output
```

Comment:
This will not work as the second selinux task will clobber the `selinux_mod_output` result from the first one.